### PR TITLE
Fix permission leak in sys.database_principals

### PR DESCRIPTION
**Pull request by thephantomthief**
### Description

Currently, sys.database_principals T-SQL view shows all the T-SQL users irrespective of the current user. The expectation is that current user should only be able to see the system users and users it has permission of.

This commit fixes the sys.database_principals view to align with the expected T-SQL behaviour.

Task: BABEL-4935
Signed-off-by: Sharu Goel <goelshar@amazon.com>

### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** N/A

* **Arbitrary inputs -** N/A

* **Negative test cases -** N/A

* **Minor version upgrade tests -** Already present

* **Major version upgrade tests -** Already present

* **Performance tests -** N/A

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/input/BABEL-4935.mix [hunk 1]
```diff
@@ -0,0 +1,79 @@
+-- tsql
```
Comment (shalinilohia50) [inline test/JDBC/input/BABEL-4935.mix]: We should rename the test file to `sys_database_principals.mix`.

File: test/JDBC/input/BABEL-4935.mix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+-- psql
+-- Need to terminate active session before cleaning up the login
+SELECT pg_terminate_backend(pid) FROM pg_stat_get_activity(NULL)
+WHERE sys.suser_name(usesysid) = 'babel_4935_sysadmin' AND backend_type = 'client backend' AND usesysid IS NOT NULL;
+GO
+
+SELECT pg_terminate_backend(pid) FROM pg_stat_get_activity(NULL)
+WHERE sys.suser_name(usesysid) = 'babel_4935_no_sysadmin1' AND backend_type = 'client backend' AND usesysid IS NOT NULL;
+GO
```
Comment (shalinilohia50) [inline test/JDBC/input/BABEL-4935.mix]: We should do the same for login `babel_4935_no_sysadmin2`.

Comment (thephantomthief) [reply test/JDBC/input/BABEL-4935.mix]: We actually never create a connection using babel_4935_no_sysadmin2. It is just created to ensure non-sysadmin connection only see their own user and not any other logins' users.

File: contrib/babelfishpg_tsql/sql/ownership.sql [hunk 3]
```diff
@@ -423,6 +423,8 @@ ON Base.rolname = Ext.rolname
 LEFT OUTER JOIN pg_catalog.pg_roles Base2
 ON Ext.login_name = Base2.rolname
 WHERE Ext.database_name = DB_NAME()
+  AND (Ext.orig_username IN ('dbo', 'db_owner', 'guest', 'sys', 'INFORMATION_SCHEMA', 'public', 'sys') -- system users should always be visible
+  OR pg_has_role(Ext.rolname, 'MEMBER')) -- Current user should be able to see users it has permission of
```
Comment (HarshLunagariya) [inline contrib/babelfishpg_tsql/sql/ownership.sql]: Can we remove INFORMATION_SCHEMA, public, sys from this list? Looks like they are covered in other part of UNION.

File: test/JDBC/expected/BABEL-4935.out [hunk 4]
```diff
@@ -0,0 +1,144 @@
+-- tsql
```
Comment (HarshLunagariya) [inline test/JDBC/expected/BABEL-4935.out]: Let's add these tests in any existing test file.

**Review by HarshLunagariya — CHANGES_REQUESTED** [review]

File: test/JDBC/input/BABEL-4935.mix (hunk 1 above)
Comment (thephantomthief) [reply test/JDBC/input/BABEL-4935.mix]: Added these test to an existing file.

**Review by HarshLunagariya — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**